### ISSUE TITLE
Bug with select  from space with uuid type field by SQL protocol

### ISSUE THREAD

**Issue by chelsEg** [tarantool-php/client#76]
I'm detect some interest case:

Create **example** space:
```
uuid = require('uuid')

local space = box.schema.space.create('example')
space:create_index('primary', {type = 'tree', parts = {1, 'unsigned'}}) 
space:format({
    {name = 'id', type = 'unsigned'}, 
    {name = 'uuid_field', type = 'uuid'}, 
    {name = 'ts', type = 'unsigned'}
})

space:insert({1, uuid.fromstr('f0fcc1c2-1740-4275-abdc-40ad5df53785'), 1638452366})
```

Get data by **binary protocol**:

```
$result = $client()->getSpace('example')->select(Criteria::key([$id]));
var_dump($result);

Array
(
    [0] => Array
        (
            [0] => 1
            [1] => Symfony\Component\Uid\UuidV4 Object
                (
                    [uid:protected] => f0fcc1c2-1740-4275-abdc-40ad5df53785
                )

            [2] => 1638452366
        )
)
```

Get data by **SQL protocol**:

```
$result = $client()->executeQuery('select * from "example" where "id" = ?', $id);
var_dump($result);

Tarantool\Client\SqlQueryResult Object
(
    [data:Tarantool\Client\SqlQueryResult:private] => Array
        (
            [0] => Array
                (
                    [0] => 1
                    [1] => �����@Bu��@�]�7�
                    [2] => 1638452366
                )

        )

    [metadata:Tarantool\Client\SqlQueryResult:private] => Array
        (
            [0] => Array
                (
                    [0] => id
                    [1] => unsigned
                )

            [1] => Array
                (
                    [0] => uuid_field
                    [1] => uuid
                )

            [2] => Array
                (
                    [0] => ts
                    [1] => unsigned
                )

        )

    [keys:Tarantool\Client\SqlQueryResult:private] => Array
        (
            [0] => id
            [1] => uuid_field
            [2] => ts
        )
)
```

The field value **uuid_field** looks like a set of random characters.

I think it's a bug.

**Comment by rybakit** [comment]
@chelsEg What Tarantool version are you on? I can't reproduce your bug on the latest Tarantool v2.10.0-beta1.

**Comment by rybakit** [comment]
I dug a little deeper and found out that it is a Tarantool issue. Tarantool >= 2.4, < 2.10.0-beta1 returns uuid data as `MP_BIN`, so client cannot unpack it into a Uuid object: https://github.com/tarantool-php/client/runs/4403128540?check_suite_focus=true.

Tarantool 2.10.0-beta1 started to return the correct `MP_EXT`, which the client can unpack into a Uuid object: https://github.com/tarantool-php/client/runs/4403129152?check_suite_focus=true.

However, there seems to be another problem, Tarantool cannot find the record if uuid is passed in the WHERE clause. This test does not work, but it should (https://github.com/tarantool-php/client/runs/4402887035?check_suite_focus=true):

https://github.com/tarantool-php/client/blob/44db2c980ca2f2ae8664360c8a6f6ac891ff4efb/tests/Integration/MessagePack/UuidExtensionTest.php#L51-L60

**Comment by chelsEg** [comment]
> @chelsEg What Tarantool version are you on? I can't reproduce your bug on the latest Tarantool v2.10.0-beta1.

Forgot say about this.

I'm used Tarantool 2.8.2

**Comment by rybakit** [comment]
@chelsEg I am closing this issue as it is not related to the client, but to Tarantool. Please vote on [this ticket](https://github.com/tarantool/tarantool/issues/6766) if you are interested in fixing the issue more quickly. For my part, I added a couple of tests ([uuid](https://github.com/tarantool-php/client/blob/aa2e70cc766a26aec75815bf7653fe8916385877/tests/Integration/MessagePack/UuidExtensionTest.php#L51-L63), [decimal](https://github.com/tarantool-php/client/blob/aa2e70cc766a26aec75815bf7653fe8916385877/tests/Integration/MessagePack/DecimalExtensionTest.php#L54-L66)) to make sure that the client works as expected on future versions of Tarantool (I marked the new tests to run only on >=2.10 stable, so hopefully the issue will be fixed by the 2.10 release).
